### DISCUSSION TITLE
Release 1.4.0/1.5.0 Possible Schedule Changes

### DISCUSSION THREAD

**Discussion by Firstyear** [kanidm/kanidm#3092] [Polls]
Our upcoming release dates are:

* November 1st 2024 (1.4.0)
* February 1st 2025 (1.5.0)

In this time, there is a possibility that I may be presenting a live demo of Kanidm at a conference here in Australia. This would be around Jan 20th 2025.

Now because of this, there is a high likelihood of an influx of new users to the project. I really want to make sure we have the best demo possible, and I want to demo a released version so that people can immediately use what they see. Because of this I want to be sure we polish some of our newer features before then.  This would mean that the version likely to be displayed is 1.4.0 (if we stick to the current schedule), where we have ~3 weeks remaining for this.

**Comment by tumbl3w33d** [comment]
I would always prefer releasing more often and sooner. Fast feedback cycle, integrate often, keep the gap with real world use small.

If you want a compromise for those afraid of changing version numbers, you could declare 1.3 as LTS (aka security fixes being backported, feature set frozen) and simply move faster starting with 1.4. People can then jump on the train which has a velocity they are comfortable with.

**Reply by CEbbinghaus** [reply]
I do like this. Having frequent updates also removes a lot of the mental burden of "have to get this feature done". A lot less stressful if there are more releases so the next one is only a month out not 3.

With any luck this also extends to users updating kanidm more regularly so people don't sit on old version for too long (unrealistic but I can dream)

**Reply by Firstyear** [reply]
I think you all missed the point - this wasn't about long term, I'm happy with the 3 month schedule.

my point was we have a major public display upcoming, and I wanted to know how to handle that. I think rn given our cycle, it'll be 1.4.6, and that's fine, then we return to "normalicy".

**Reply by CEbbinghaus** [reply]
that was what I thought. Just wanted to also voice my support for more frequent release cycles (or at least minor bump cycle. Patches are released frequently enough as is)

**Comment by yaleman** [comment]
Do what we have time and sanity for. 😂
